packages/rulesets/src/spectral/functions/properties-type-object-no-definition.ts

```diff
@@ -0,0 +1,15 @@
+// If a property is defined with `'type': 'object'`, it must also define an object body.
+// RPC Code: RPC-Policy-V1-03
+const errorMessage =
+  "If Properties with type:object dont have a reference model defined, then the allowed types can only be primitive data types."
+export const propertiesTypeObjectNoDefinition: any = (definitionObject: any, _opts: any, ctx: any) => {
+  if (definitionObject?.type === "") {
+    return [{ message: errorMessage, path: ctx.path }]
```

Comment:
>errorMessage [](http://example.com/codeflow?start=23&length=12)

is this code reachable? The caller of this method is already checking that @property === 'type' && @ ==='object' so in line number 7 here type can never be empty right?

Reply:
This code is reachable when type:''   is left empty.
The given expression I provided will give the result of type:''

Reply:
got it! my bad, was reading the given condition wrong